packages/babel-plugin-proposal-do-expressions/src/index.js

```diff
@@ -11,7 +11,12 @@ export default declare(api => {
     visitor: {
       DoExpression: {
         exit(path) {
-          const body = path.node.body.body;
+          const { node } = path;
+          if (node.async) {
+            // Async do expressions are not yet supported
+            return;
```

Comment:
Nit: `path.buildCodeFrameError`?

Reply:
I lean to skip these paths otherwise if we support them in `proposal-async-do-expressions`, the error here will impose a restriction that `proposal-async-do-expressions` must run before `proposal-do-expressions`.

We have similar approach on skipping `for await of` in `for-of` transform: https://github.com/babel/babel/blob/16d83002defc74dcbf0d73d8b71e9646acf04f16/packages/babel-plugin-transform-for-of/src/index.js#L58-L60